go/dism/dism.go

```diff
@@ -147,17 +153,31 @@ func (s Session) RemovePackage(
 	cancelEvent *windows.Handle,
 	progressCallback unsafe.Pointer,
 ) error {
-	return DismRemovePackage(s.Handle, helpers.StringToPtrOrNil(identifier), packageIdentifier, cancelEvent, progressCallback, nil)
+	return s.checkError(DismRemovePackage(*s.Handle, helpers.StringToPtrOrNil(identifier), packageIdentifier, cancelEvent, progressCallback, nil))
 }
 
 // Close closes the session and shuts down dism. This must be called prior to exiting.
 func (s Session) Close() error {
-	if err := DismCloseSession(s.Handle); err != nil {
+	if err := DismCloseSession(*s.Handle); err != nil {
 		return err
 	}
 	return DismShutdown()
 }
 
+// checkError validates the error returned by DISM API and reloads the session if needed
+func (s Session) checkError(err error) error {
+	if err == DISMAPI_S_RELOAD_IMAGE_SESSION_REQUIRED {
```

Comment:
I think it would be ideal to have some basic logging in place here for debugging purposes, to surface the fact that the session is being modified. It looks like we haven't added logging to this package yet, but at minimum please add a TODO here for us to incorporate some log events. (We use logger in the other glazier packages if you want to DIY).